### PR TITLE
bin: make raft ticks configurable

### PR DESCRIPTION
**Pull request by BusyJay**
It's convenient to be able to configure ticks without recompiling the source.

@siddontang @hhkbp2 PTAL

### CODE REVIEW

File: etc/config-template.toml [hunk 1]
```diff
@@ -38,6 +38,18 @@ address = ""
 job = "tikv"
 
 [raftstore]
+# Base tick interval
```
Comment (BusyJay) [inline etc/config-template.toml]: There is no reliable way to check this right now.

File: src/bin/tikv-server.rs [hunk 2]
```diff
@@ -112,6 +112,38 @@ fn get_boolean_value<F>(short: &str,
         .expect(&format!("please specify {}", long))
 }
 
+// A helper macro that generate functions that can read int from configuration and
+// than transform into other type.
+macro_rules! cfg_int_func {
+    ($t:ty, $name:ident, $i:ident => $trans:expr) => (
```
Comment (siddontang) [inline src/bin/tikv-server.rs]: seem we can use function instead of macro.

File: etc/config-template.toml [hunk 3]
```diff
@@ -38,6 +38,18 @@ address = ""
...
+# raft-base-tick-interval = 100ms
+
+# Ticks to trigger a hearbeat
+# raft-heartbeat-ticks = 3
+
+# Ticks that must pass between elections. It must be greater than
```
Comment (hhkbp2) [inline etc/config-template.toml]: For good user experience, docs should also be provided for

1. the brief description of tick, heartbeat, election timeout, etc
2. the principle of how to configure them correctly, pros and cons, example cases

We could mark a TODO here or somewhere in the task management system, to recap the corresponding docs/guide in the future.

File: etc/config-template.toml (hunk 1 above)
Comment (hhkbp2) [inline etc/config-template.toml]: A log out could be helpful for info/debug. Yet it works only for normal users, not against system crackers.

File: src/raftstore/store/config.rs [hunk 4]
```diff
@@ -116,7 +116,7 @@ impl Default for Config {
             snap_mgr_gc_tick_interval: DEFAULT_MGR_GC_TICK_INTERVAL_MS,
             snap_gc_timeout: DEFAULT_SNAP_GC_TIMEOUT_SECS,
             messages_per_tick: DEFAULT_MESSAGES_PER_TICK,
-            max_peer_down_duration: Duration::from_secs(DEFAULT_MAX_PEER_DOWN_SECS),
+            max_peer_down_duration: Duration::from_millis(DEFAULT_MAX_PEER_DOWN_SECS),
```
Comment (hhkbp2) [inline src/raftstore/store/config.rs]: s/..._SECS/..._MS/ ?

**Comment by BusyJay** [comment]
The option should be public after there is a reliable way to validate it.

**Pull request closed without merging**